### DISCUSSION TITLE
85. 트랜잭션 격리 수준에서 발생할 수 있는 3가지 데이터 부정합 현상에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#87] [General]
키워드: Dirty Read, Non-Repeatable Read, Phantom Read

**Comment by linirini** [comment]
- Dirty Read
  - 한 트랜잭션에서 커밋 되지 않은 수정 중인 작업 사항을 다른 트랜잭션이 읽을 수 있게 되는 현상
  - 발생 레벨: Read Uncommitted
- Non-Repeatable Read
  - 한 트랜잭션 내에서 동일한 쿼리를 두 번 수행하는데, 그 사이에 커밋된 다른 트랜잭션의 변경 사항에 의해 두 번 모두 다른 결과가 나오는 현상
  - 발생 레벨: Read Committed, Read Uncommitted
- Phantom Read
  - 한 트랜잭션에서 일정 범위의 데이터를 두 번 이상 읽을 때 없던 데이터가 나타나거나 있던 데이터가 사라지는 현상
  - 발생 레벨: Repeatable Read, Read Committed, Read Uncommitted
